x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/BlobCacheMetrics.java

```diff
@@ -56,4 +103,47 @@ public LongCounter getEvictedCountNonZeroFrequency() {
     public LongHistogram getCacheMissLoadTimes() {
         return cacheMissLoadTimes;
     }
+
+    /**
+     * Record the various cache population metrics after a chunk is copied to the cache
+     *
+     * @param totalBytesCopied The total number of bytes copied
+     * @param totalCopyTimeNanos The time taken to copy the bytes in nanoseconds
+     * @param shardId The shard ID to which the chunk belonged
+     * @param cachePopulationReason The reason for the cache being populated
+     */
+    public void recordCachePopulationMetrics(
+        int totalBytesCopied,
+        long totalCopyTimeNanos,
+        ShardId shardId,
+        CachePopulationReason cachePopulationReason
+    ) {
+        Map<String, Object> metricAttributes = Map.of(
+            SHARD_ID_ATTRIBUTE_KEY,
+            shardId.toString(),
+            CACHE_POPULATION_REASON_ATTRIBUTE_KEY,
+            cachePopulationReason.name()
+        );
```

Comment:
I changed these values to be strings, because it looks like that's what `OtelHelper.fromMap` (Ultimately used by `APMMeterRegistry`) expects. Recording these metrics seems slightly expensive?

Reply:
In other places, e.g. recovery metrics, we are recording attributes for shardId as two fields of "index_name" and "shard_id" (the integer ID of the shard). I think we should do the same here.

Reply:
That makes much more sense. Thanks, updated.